src/core/tests/frontend/shared/src/json_config_extension.cpp

```diff
@@ -0,0 +1,194 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "json_config_extension.hpp"
+
+#include <json_extension/json_config_extension.hpp>
+#include <json_extension/json_schema.hpp>
+#include <json_extension/json_transformation_extension.hpp>
+#include <nlohmann/json-schema.hpp>
+#include <ostream>
+#include <util/graph_comparator.hpp>
+
+#include "utils.hpp"
+
+using namespace ov::frontend;
+
+std::string FrontEndJsonConfigTest::getTestCaseName(const testing::TestParamInfo<JsonConfigFEParam>& obj) {
+    std::string res = obj.param.m_frontEndName + "_" + obj.param.m_modelName;
+    return FrontEndTestUtils::fileToTestName(res);
+}
+
+void FrontEndJsonConfigTest::SetUp() {
+    FrontEndTestUtils::setupTestEnv();
+    m_fem = FrontEndManager();  // re-initialize after setting up environment
+    initParamTest();
+}
+
+void FrontEndJsonConfigTest::initParamTest() {
+    m_param = GetParam();
+    m_param.m_modelName = FrontEndTestUtils::make_model_path(m_param.m_modelsPath + m_param.m_modelName);
+}
+
+inline std::string get_lib_path(const std::string& lib_name) {
+    return ov::util::make_plugin_library_name<char>(ov::util::get_ov_lib_path(), lib_name + IE_BUILD_POSTFIX);
+}
+
+std::string generate_json_config() {
+    std::string json = R"(
+    [
+      {
+        "custom_attributes": {
+          "test_attribute": true
+        },
+        "id": "buildin_extensions_1::TestExtension1",
+        "library": "test_builtin_extensions_1",
+        "match_kind": "scope"
+      },
+      {
+        "custom_attributes": {
+          "test_attribute": true
+        },
+        "id": "buildin_extensions_2::TestExtension1",
+        "library": "test_builtin_extensions_2",
+        "match_kind": "scope"
+      },
+      {
+        "custom_attributes": {
+          "test_attribute": true
+        },
+        "id": "buildin_extensions_2::TestExtension2",
+        "library": "test_builtin_extensions_2",
+        "match_kind": "scope"
+      }
+    ]
+    )";
+    nlohmann::json config_json = nlohmann::json::parse(json);
+
+    // Validate JSON config
+    nlohmann::json_schema::json_validator validator;
+    validator.set_root_schema(json_schema);
+    validator.validate(config_json);
+
+    for (auto& section : config_json) {
+        section["library"] = get_lib_path(section["library"]);
+    }
+    std::string path = "/home/itikhonov/OpenVINO/openvino/src/core/tests/frontend/shared/include/json_files/test.json";
```

Comment:
local path

Reply:
replaced with "test_json.json" created in local dir
added std::remove in TearDown